### ISSUE TITLE
Spike: Investigate jest-axe

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#21961]
## Description
We have decided that our current playwright tests are not providing a lot of benefits and we are considering removing them. However, we want to having accessibility testing for a few reasons:

1. We want to be alignment with the rest of the platform (they use cypress-axe)
2. The VA/OCTO/the project/I have all done quite a bit to make the pages we are working on quite accessible, we do not want to go backwards and so we want to make sure that we set the repo up for success for folks that come after us
3. We want to make people accountable for writing good, accessible code. Having basic axe tests is an easy way to catch simple issues before they go to production

We would essentially want to come away from this spike with a clear understanding of what can we easily do, what could we do with some effort, and what could we not do (what isn't possible).

### Specifics
The [content-build](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/platform/testing/e2e/cypress/support/commands/axeCheck.js) and [vets-website](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/testing/e2e/cypress/support/commands/axeCheck.js) have specific testing. This is dictated by Platform and we should aim to match this. (looking just now it seems like the content build rules are out of date so we should match vets-website)

**Rules**
- We test against Section 508, WCAG 2.0 level A & AA, WCAG 2.1 level A & AA
- We do NOT test against color contrast
- We have recently enabled heading rules:
  - Testing for empty headings
  - Testing for heading order
  - Testing for 1 H1

**Other**
- Ideally we would also want to test PRs before merge, so that violations are found before they make it before they go to production.
- We also have to at least consider the shadow DOM 

## Acceptance Criteria
- [ ] Investigate testing using jest-axe
- [ ] Document what can we easily do, what could we do with some effort, and what could we not do (what isn't possible).
